### ISSUE TITLE
marker_mark colors not taking effect

### ISSUE THREAD

**Issue by mscdex** [geany/geany#916]
I'm trying to customize an existing theme but am finding that the `marker_mark` colors are not being used (completely). For example, if I try to set `marker_mark=#000;#FFCC00`, I would expect black on bright yellow text. Instead I'm seeing the original foreground color of the marked text with the background color being a dark yellow.

I've even tried to change the `marker_transparency` settings but that didn't seem to make any difference at all.

I am using geany v1.26 on Linux.

Any ideas as to what I'm doing wrong?

**Comment by b4n** [comment]
`marker_mark` is used for colors of the markers in the marker margin, not in the text.  What you see is probably the `marker_search`, for which only the background is used.  See http://www.geany.org/manual/#id4 and the values there.

**Comment by mscdex** [comment]
I've actually tried changing that value as well, the background color is still a dark yellow.

**Comment by b4n** [comment]
Ok, so we'll need to know what marker it really is (how was it created) and maybe what theme you're trying to modify.

**Comment by elextr** [comment]
Just to check, you did re-load the colour scheme?  Geany won't do that automatically AFAIK

**Comment by mscdex** [comment]
I copied the Darcula theme, changed some of the theme_info to differentiate it from Darcula, and simply changed the existing `marker_search` to `marker_search=;#ffff00`.

I've also noticed if I try to use any normally bright color (for example `#00ff00`), it's just a dark version of the color, even though the brightness is correct for other syntax highlighting options (e.g. when matching a good bracket).

@elextr Yes I am doing that. I load the theme selector and click back and forth to reload. I have also tried restarting Geany.

**Comment by elextr** [comment]
Search marking:

1. only changes the background, so as @b4n said the foreground setting is ignored, this is the way Scintilla (the editing component we use) works, it means you can still see syntax styling on the marked text

2. is applied with a fixed alpha of 60, so you will never get your bright colours

**Comment by mscdex** [comment]
Is it possible to make the alpha level configurable?

**Comment by elextr** [comment]
Sure, just "somebody" has to do it.

**Comment by codebrainz** [comment]
@mscdex maybe add third field to [`translucency`](https://github.com/geany/geany/blob/master/data/filetypes.common#L29)? Just copy the way the existing two work and make the 3rd field get passed [here](https://github.com/geany/geany/blob/master/src/highlighting.c#L658) instead of 60, I think.

**Comment by elextr** [comment]
@codebrainz yeah or [here](https://github.com/geany/geany/blob/master/data/filetypes.common#L42) to continue the confusion between "markers" and "marks" :)

**Comment by mscdex** [comment]
FWIW I just tried changing the hardcoded value of 60 to any other value (including 255 for no transparency) and it didn't make any difference at all.

Just to be clear, I'm testing with the mark you get when you highlight a token/word and press ctrl+shift+m to highlight all instances of that token/word.

**Comment by mscdex** [comment]
Hrmm... I must not be doing something right.... I even hardcoded the `SCI_INDICSETFORE` value and it still uses the color defined in my theme.

**Comment by elextr** [comment]
Maybe post a gist with a diff of what you changed

**Comment by mscdex** [comment]
@elextr gist: https://gist.github.com/mscdex/35ccffa0e10beb93fbd8

**Comment by elextr** [comment]
Hmmm, well, assuming it `configure`, `make` and `make install`ed correctly and you are running the modified version (yeah I've run the wrong one :) it should do what the hard coding says AFAICT.

**Comment by mscdex** [comment]
Yep it is the right one. The system copy is in `/usr/bin` and the one I'm testing is in `/usr/local/bin`. If someone wouldn't mind testing the same change when they get some time I would appreciate it. That way I can know if it's just me or what.

**Comment by elextr** [comment]
I'm afraid it appears to be just you :(

Works for me :)

Don't know what you are doing different, my recipe:

```
mkdir /path/xxx; cd /path/xxx; git clone https://github.com/geany/geany.git; cd geany
./autogen.sh --prefix=/path/xxx; make install; cd ../bin; ./geany -c ../config
```

That doesn't touch any system dirs (note no sudo needed :)

**Comment by elextr** [comment]
> The system copy is in /usr/bin and the one I'm testing is installed to /usr/local/bin.

The geany executable is just a miniscule thing that links to libgeany, so my first guess is you might be getting the wrong libgeany, IIUC the dynamic loader looks in some system dirs like `/usr/lib` and `/usr/local/lib` specially so that might be confusing it.

**Comment by b4n** [comment]
The Gist WFM just fine, *but* note that any value outside the 0-255 (included) range will be ignored, plain and simple.
Opened PR #917 fixing the documentation and adding a setting for the search marker translucencies.

**Comment by mscdex** [comment]
@elextr I wasn't aware of that, I will look at it again.

**Comment by SweaterFish** [comment]
The commits added in 2018 did not resolve this issue.

After a bunch of fruitless testing, I found that what was added was a setting in filedefs.common

```
# translucency for the line marker(first argument) and the search marker (second argument)
marker_translucency=256;256

```

However, this only allows setting translucency for marker_line and marker_mark, not marker_search as the description implies and was the issue reported here. This does make it marker_search highlighting almost unusable with dark schemes.

Would anyone care to add the third, and originally requested, marker translucency setting to this line? It should be easy, but I'm not familiar with Geany source at all so it would be better if someone else did it.

Additionally, this description is poor. Why use ambiguous descriptions like "the line marker" and "the search marker" instead of just saying "marker_line" and "marker_mark"?

Additionally additionally, it would be nice if these translucencies could be (optionally) defined in the colorscheme config instead of the separate filedefs.common. For other definitions in filedefs.common, you can just have, e.g.

```
marker_search=marker_search
```

to use whatever was set in the colorscheme definition file. The same does not appear to work for either marker_transparency or the other transparency setting further up (that affects current_line and selection).
